sci-libs/openblas/openblas-0.3.17.recipe

```diff
@@ -61,11 +62,11 @@ case "$effectiveTargetArchitecture" in
 	*) target=;;
 esac
 
-BUILD_CONF="NO_STATIC=1 NO_LAPACK=1 \
+BUILD_CONF="NO_STATIC=1 \
+		NO_LAPACK=1 \
 		NO_LAPACKE=1 \
 		NO_AFFINITY=1 \
 		NO_WARMUP=1 \
-		NO_AVX=1 \
```

Comment:
AVX is only on x86_64 available. NO_AVX=1 should stay for x86.

Reply:
This should be detected during Build on x86.  https://github.com/xianyi/OpenBLAS/blob/1a8b6134c24f66c649ca342c74d551055b98cbc6/Makefile.system  - 32-bit binaries should have these flags added by the build system

Have you tried building it on x86?

Reply:
This won't be detected during build.

Reply:
I don't understand why.  Are you saying the 32 bit build system will not be detected and then somehow try to build AVX features?  The build process should use the provided GCC to detect whether it can generate 32-bit vs 64-bit.  It sets the BINARY env variable which then, if it's = 32 will provide the NO_AVX, etc. itself.

NO_AVX is implicit for 32 bit binaries.  It should not need to be provided separately... If so it should be NO_AVX, NO_AVX2, NO_AVX512, etc.

Reply:
I mean the build won't detect that 32-bit Haiku has no AVX support. If OpenBLAS disables AVX for every 32-bit system (which would be worrying), then OK.

Reply:
32 bit OpenBLAS has no AVX support, it's not possible anyway:

```
ifeq ($(BINARY), 32)
GETARCH_FLAGS	+= -DNO_AVX -DNO_AVX2 -DNO_AVX512
NO_AVX512 = 1
endif
...
ifeq ($(ARCH), x86)
CCOMMON_OPT	+= -DNO_AVX
endif
```

Reply:
So it's not detected, but hardcoded disabled. Should be OK.